flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java

```diff
@@ -376,12 +410,13 @@ private JobGraph createBatchJobGraph() {
 	}
 
 	private ExecutionGraph createExecutionGraph(final JobGraph jobGraph) throws Exception {
-		return createExecutionGraph(jobGraph, new FixedDelayRestartStrategy(10, 0));
+		return createExecutionGraph(jobGraph, new FixedDelayRestartStrategy(10, 0), false);
 	}
 
 	private ExecutionGraph createExecutionGraph(
 			final JobGraph jobGraph,
-			final RestartStrategy restartStrategy) throws Exception {
+			final RestartStrategy restartStrategy,
+			final boolean failOnRecoverCheckpoint) throws Exception {
```

Comment:
Agreed, combine `testFailGlobalIfErrorOnResetTasks` and previous `testFailGlobalIfErrorOnRescheduleTasks` into one unit test as `testFailGlobalIfErrorOnRestartTasks` with given RestartStrategy which would fail when restarting.